src/plugins/unified_search/public/query_string_input/query_bar_top_row.tsx

```diff
@@ -486,6 +508,48 @@ export const QueryBarTopRow = React.memo(
       return <EuiFlexItem className={wrapperClasses}>{component}</EuiFlexItem>;
     }
 
+    function renderCancelButton() {
+      const buttonLabelCancel = strings.getCancelQueryLabel();
+
+      if (submitButtonIconOnly) {
+        return (
+          <EuiButtonIcon
+            iconType="cross"
+            aria-label={buttonLabelCancel}
+            onClick={onClickCancelButton}
+            size={shouldShowDatePickerAsBadge() ? 's' : 'm'}
+            data-test-subj="queryCancelButton"
+            // @ts-expect-error Need to fix expecting `children` in EUI
+            toolTipProps={{
+              content: buttonLabelCancel,
+              delay: 'long',
+              position: 'bottom',
+            }}
+          >
+            {buttonLabelCancel}
+          </EuiButtonIcon>
+        );
+      }
+
+      return (
+        <EuiButton
+          iconType="cross"
+          aria-label={buttonLabelCancel}
+          onClick={onClickCancelButton}
+          size={shouldShowDatePickerAsBadge() ? 's' : 'm'}
+          data-test-subj="queryCancelButton"
```

Comment:
```suggestion
          data-test-subj="queryCancelButton"
          color="danger"
```
I'd also recommend using the `danger` colour for this button too to distinguish it from the refresh button.